common/reviews/api/rush-lib.api.md

```diff
@@ -429,13 +499,19 @@ export interface IOperationRunner {
 
 // @beta
 export interface IOperationRunnerContext {
+    readonly changedProjectsOnly: boolean;
     collatedWriter: CollatedWriter;
+    readonly consumers: Set<IOperationRunnerContext>;
```

Comment:
Maybe the reason of this change comes from the fact that I am not fully understanding the difference between `OperationExecutionRecord` and `IOperationRunnerContext`.  I can see that `consumers` is one of the properties of `OperationExecutionRecord`... Could you kindly elaborate a little bit more about his? It would be helpful if i can change the code to not be that "strange" :)

Reply:
IOperationRunnerContext is the set of data the operation runner (which should only ever be concerned with the current operation) needs to know. OperationExecutionRecord is the persistent data structure for the state of an operation and its relationship to its dependencies and consumers.

This relationship would get fixed by using the operation-level "beforeExecuteOperation" and "afterExecuteOperation" hooks to handle the cache interaction instead of bizarrely hooking inside of ShellOperationRunner (it also solves the log lifecycle issue).